File: Python/Examples/solar_tracker.py
import os             
import sys
cwd=os.getcwd()                    # Gets the current working directory of the file
(setpath,Examples)=os.path.split(cwd)   # Creates a tuple (head,tail) where tail is everything after the final slash
print setpath                    
sys.path.append(setpath)              #Appends the setpath to sys.path so as to provide a search path for the module Arduino

from Arduino.Arduino import Arduino        # Imports all the functions from module Arduino
from time import sleep                      

class SOLAR_TRACKER:
	def __init__(self,baudrate):
		self.baudrate=baudrate         
		self.setup()
		self.run()
		self.exit()

	def setup(self):
		self.obj_arduino=Arduino()
		self.port=self.obj_arduino.locateport()          #Locates the port 
		self.obj_arduino.open_serial(1,self.port,self.baudrate)

	def run(self):
		self.ldr1=0                      #LDR1 connected at port 0
		self.ldr2=1                      #LDR2 connected at port 1
		s=0
		self.angle=5                   # Step angle
		self.initial=90
		self.obj_arduino.cmd_servo_attach(1,1)     # Attach the servo at pin 9
		sleep(1)
		self.obj_arduino.cmd_servo_move(1,1,self.initial)     #Rotate servo by 90 deg
		sleep(1)
		for i in range(20):
			p=self.obj_arduino.cmd_analog_in(1,self.ldr1)      #Read analog input from LDR1
			q=self.obj_arduino.cmd_analog_in(1,self.ldr2)      # Read analog input from LDR2  
			print p
			print q
			sleep(1)
			if int(p) < int(q):             
				s=s+1
				self.obj_arduino.cmd_servo_move(1,1,90-self.angle*s)   # Clockwise rotation
				sleep(1)
			else:
				s=s-1
				self.obj_arduino.cmd_servo_move(1,1,90-self.angle*s)  # Anti-clockwise rotation
				sleep(1)
		self.obj_arduino.cmd_servo_detach(1,1)		# Detach the motor

	def exit(self):
		self.obj_arduino.close_serial()           # Close the serial port 

def main():
	obj_ldr=SOLAR_TRACKER(115200)

if __name__=='__main__':                                  # Execute the module as script file
	main()

File: Julia/examples/SolarTracker.jl
using SerialPorts
using ArduinoTools

ser = connectBoard(115200)
ServoAttach(ser,1)
sleep(1)
j = 0
k = 0
r = 0
ServoMove(ser,1,90)  # Initially move the servo motor by 90 deg
for i = 1:30
  p = analogRead(ser,0)   # Read the analog value of LDR1
  println(p)
  q = analogRead(ser,1)  # Read the analog value of LDR2
  println(q)
  k = p-q  # threshold
  r = q-p  # threshold
  if p<q && r>10
    j = j+1
    ServoMove(ser,1,90-5j)   # Clockwise rotation
    sleep(0.5)
  end
  if p>q && k>10
    j = j-1
    ServoMove(ser,1,90-5j)  # Anti-clockwise
    sleep(0.5)
  end
  sleep(1)
end
ServoDetach(ser,1)
close(ser)
